### ISSUE TITLE
ege-webservice Error 400 while converting docx or xlsx documents

### ISSUE THREAD

**Issue by jthomae1** [TEIC/oxgarage#35]
We encountered another problem regarding DOCX and XLSX documents. 
In the UI, everything works as expected. But using the ege-webservice, several conversions are erroneous.

We are using a java client for oxgarage to convert a range of document types via the ege-webservie api. Most of the conversions are successful.

Affected conversions we encountered:
DOCX to PDF
DOCX to DOC
DOC to DOCX

XLSX to PDF
XLSX to XLS
XLS to XLSX

The conversion of PPTX to PDF is working, I guess only conversion cases which include XLSX and DOCX files are affected.

I tried all conversions against our own instance and [https://oxgarage2.tei-c.org/](https://oxgarage2.tei-c.org/) because we are using the same docker image.

I also tried a few conversions at [https://oxgarage.tei-c.org/](https://oxgarage.tei-c.org/) but the result was the same.

The api creates and returns the targetfile, but there only is an error message inside the document, which can be opened with a texteditor. 

`<!doctype html><html lang="en"><head><title>HTTP Status 400 – Bad Request</title><style type="text/css">H1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : black;}HR {color : #525D76;}</style></head><body><h1>HTTP Status 400 – Bad Request</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Description</b> The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).</p><hr class="line" /><h3>Apache Tomcat/7.0.94</h3></body></html>`

My first intention is not a client error, because we are handling all request the same. And all other conversions are successful. 

Maybe you can check the logs of [https://oxgarage2.tei-c.org/](https://oxgarage2.tei-c.org/) if there is any regarding information. I tried all conversations in the last 60 minutes.

If I can provide any further information, please let me know.

**Comment by peterstadler** [comment]
hmm, strange … 
Looking at the logs didn't reveal anything for me but. Playing around showed that indeed nothing was written to the logs because files > 1m were already blocked by the nginx load balancer (with an `413 Request Entity Too Large` error). I now set the limit to 50m and DOCX to PDF worked for the test files I used. 

Still strange, because the main oxgarage was not affected by this limitation?

Could you send me one of your files for testing if the error still occurs?

**Comment by jthomae1** [comment]
[Lorem.docx](https://github.com/TEIC/oxgarage/files/3317132/Lorem.docx)

That is the document I used, just a few lines of Lorem Ipsum.

I just tried it again with this file on both servers ([https://oxgarage2.tei-c.org/](https://oxgarage2.tei-c.org/) and [https://oxgarage2.tei-c.org/](https://oxgarage2.tei-c.org/)) but I'm still getting the same file with the Status 400 message inside. The conversion type was DOCX to PDF.

And that's the way I build the request in my client:

```
public InputStream convert(InputStream input, String filename, FILETYPE inputType, FILETYPE outputType) throws IOException {
        // check filetypes
        LOGGER.info("inputtype: {} - outputtype: {}", inputType, outputType);
        OxUtil.checkFileTypes(inputType, outputType);
        
        HttpClient client = HttpClientBuilder.create().build();

        String requestUrl = url.concat(CONVERSION_PATH).concat("/").concat(inputType.getPath()).concat("/").concat(outputType.getPath());
        LOGGER.info("requesturl: {}", requestUrl);
        
        MultipartEntityBuilder builder = MultipartEntityBuilder.create();
        builder.setMode(HttpMultipartMode.BROWSER_COMPATIBLE);
        builder.addPart("upfile", new InputStreamBody(input, filename));

        HttpPost post = new HttpPost(requestUrl);
        HttpEntity entity = builder.build();
        post.setEntity(entity);
        HttpResponse response = client.execute(post);
        
        return response.getEntity().getContent();
}
```

**Comment by peterstadler** [comment]
It looks like the request URL is incomplete. You'll need the full conversion path `https://oxgarage.tei-c.org/ege-webservice/Conversions/docx%3Aapplication%3Avnd.openxmlformats-officedocument.wordprocessingml.document/TEI%3Atext%3Axml/odt%3Aapplication%3Avnd.oasis.opendocument.text/pdf%3Aapplication%3Apdf/`, see http://enrich-ege.sourceforge.net/restws.html.

In a nutshell: it's not enough to simply provide input and output format (but I had to look that up as well).

**Comment by jthomae1** [comment]
Sorry for the delay, I was on vacation.

I first tried to change the requesturl but that wasn't successful. After I tried it against `https://oxgarage.tei-c.org` and `https://oxgarage.tei-c.org` I realized that only my system still produces the bad request. After an update of the docker image (dev) it worked for me as well with all conversion types I mentioned in the first post.

I think there was a problem with the docker image, too. But now it is working fine.
Thanks for the quick help.
